kale/embed/mpca.py

```diff
@@ -23,249 +21,251 @@
 from tensorly.tenalg import multi_mode_dot
 
 
-def check_ndim(X, ndim):
-    """Check if the order/mode of the input data is consistent with the given number of order/mode
+def _check_ndim(x, n_dim):
+    """Raise error if the order of the input data is not consistent with the given order value
 
     Args:
-        X (ndarray): input data
-        ndim (int): number of order/mode expected
+        x (array-like tensor): input data
+        n_dim (int): tensor order expected
 
```

Comment:
Many may not know what tensor order is. Also, since you name it as `n_dim`, even you are not calling it order so maybe you should call it "the number of dimensions (i.e., the order) of the input tensor." I would use `n_dims` rather than `n_dim` since you used `n_components` (a lot) rather than `n_component`.

Reply:
@sz144 for the docs: "even you are not calling it **order** so maybe you should call it "the number of dimensions (i.e., the order) of the input tensor."